sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -383,7 +383,14 @@ public async virtual Task InvokeFailedArg(Exception ex)
                 ex is not TaskCanceledException &&
                 !ex.Message.Contains("The request was canceled."))
             {
-                SetFailureType(ex.Message);
+                if (ex is RequestFailedException requestFailedException)
+                {
+                    SetFailureType(requestFailedException.ErrorCode);
+                }
```

Comment:
@amnguye In this case we were actually looking for the ErrorCode in the case of `BlobAlreadyExists`, and the contents of the full response content wasn't actually required. Would this cause any other problems for other values of [DataMovementConstants.ErrorCode.CannotOverwrite](https://github.com/Azure/azure-sdk-for-net/blob/ef9e6bb7d125031fbb1a7ed25521eaa4143b9f48/sdk/storage/Azure.Storage.DataMovement/src/Shared/DataMovementConstants.cs#L142C68-L142C85)?

Reply:
//cc: @schaabs

Reply:
Correct we are just looking for `BlobAlreadyExists` (and possibly other error codes) in the future. So this wouldn't cause problems for the other cannot overwrite values.